onnxruntime/core/graph/matmul_add_fusion.h

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/graph/graph_transformer.h"
+
+namespace onnxruntime {
+
+class MatMulAddFusion : public onnxruntime::GraphTransformer {
```

Comment:
Can we try to implement this as a rewrite rule?
It is a pity to traverse the whole graph for these local transformations.
You can check the identity elimination as well as the the PR [here](https://github.com/Microsoft/onnxruntime/pull/87). It is a bit outdated but I will rebase it most probably today.